Dockerfile

```diff
@@ -78,8 +78,10 @@ RUN apt update \
   && apt install -y curl gpg libdlib19.1 ffmpeg exiftool libheif1
 
 # Install Darktable if building for a supported architecture
+# And create darktable directories for non-root users
 RUN if [ "${TARGETPLATFORM}" = "linux/amd64" ] || [ "${TARGETPLATFORM}" = "linux/arm64" ]; then \
-  apt install -y darktable; fi
+  apt install -y darktable && \
+  mkdir -p /.cache/darktable /.config/darktable; fi
```

Comment:
The `.cache` and `.config` folders are created in the user's home folder: in case, the image is running under the root, they are created in the `/root` folder. In case, it is running under a non-root user, it will be in the `/home/<user>` folder. There should be permission to create files and folders in user's home folder, so I don't see a need to do it at the image creation time (which will require also `chown` and `chmod` commands to be run, as Dockerfile is executed by root and folders, created here, are owned by root as well). It makes sense only if you also add the `USER` declaration at the end of the Dockerfile, which will result in the creation of the image, always running under specified non-root user

Reply:
but if for some reason the `darktable`, executed under non-root user, wants to create these folders in the Dockerfile's WORKDIR location, then from my point of view it makes more sense to set the RWX permissions to the folder and all its content for all users by the command `chmod -R 777 /app` - this is the application's working folder and the application has to have permissions to do anything there, even if run in non-root mode.

Reply:
I've updated my PR #863 with the changes, proposed in the previous comment. Please try to build and run the image from that PR and test whether the problem is fixed for you

Reply:
thanks for your feedback. I'll do a review of #863 and try to include the fix for this issue

Reply:
Today I did a code review of my PR #863 and rewrote it to build an image, configured to run under a non-root user.
As part of this change, I investigated the issue that Darktable fails to run because of missing 2 mentioned folders and with inability to create them. As I see, there is no need to create them in the `/` folder, as Darktable wants them to be in the user's home folder. So, once the `/home/user` folder exists and is owned by the user, under which Photoview is running, Darktable creates those folders inside it successfully and runs fine.